app/main.py

```
@@ -1 +1,36 @@
-# write your code here
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list[dict], cafe: Cafe) -> str:
+    """
+    Determines if a group of friends can visit the cafe based on access rules.
+
+    :param friends: A list of visitor dictionaries.
+    :param cafe: The Cafe object to visit.
+    :returns: A string indicating the group's status.
+    """
+    masks_to_buy = 0
+    vaccine_problem_found = False
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+
+        except NotWearingMaskError:
+            masks_to_buy += 1
+
+        except VaccineError:
+            vaccine_problem_found = True
+            break
+
+        except Exception:
+            break
```

This generic `except Exception:` block is too broad and can hide potential bugs. For example, if a `friend` dictionary is missing a required key like `"name"`, it would cause a `KeyError` in `visit_cafe`. This block would catch it, break the loop, and the function might incorrectly return a success message. It's better to only catch the specific exceptions you expect and let unexpected ones fail loudly so they can be fixed.